TripCheckins/Application flow/AppCoordinator.swift

```diff
@@ -11,19 +11,27 @@ import UIKit
 class AppCoordinator {
     
     let navigationController: UINavigationController
-    let authorizationTokenKeeper: AuthorizationTokenKeeper?
+    let authorizationTokenKeeper: AuthorizationTokenKeeper
+    let localItemsStorage: LocalItemsStorage
+    
+    private lazy var tripService = LocalTripService(localItemsStorage: localItemsStorage,
+                                                    additionalTripLoadingService: PredefinedTripService())
+    
     var foursquareAuthorizer: FoursquareAuthorizer?
     
     init(navigationController: UINavigationController,
-         authorizationTokenKeeper: AuthorizationTokenKeeper? = nil) {
+         authorizationTokenKeeper: AuthorizationTokenKeeper,
+         localItemsStorage: LocalItemsStorage) {
         self.navigationController = navigationController
         self.authorizationTokenKeeper = authorizationTokenKeeper
+        self.localItemsStorage = localItemsStorage
         
-        if let token = authorizationTokenKeeper?.authorizationToken() {
-            showCheckinsList(authorizationToken: token)
-        } else {
+        guard authorizationTokenKeeper.authorizationToken() != nil else {
```

Comment:
I think the usage of guard here is debatable. 

I see guards as similar to asserts that allow you to do something before you exit the current scope. I think the use case you have here is better served by an if. 

See https://www.natashatherobot.com/swift-when-to-use-guard-vs-if/

Reply:
Good point 👍  I actually changed this in the other PR that is currently in review so won't address the comment for now.